### PR TITLE
chore(otel_test): rework test to prep for keep id resource attributes

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Rework the test to be more explicit and support testing for https://github.com/prometheus/prometheus/pull/15448 that's coming next.

#### Which issue(s) this PR fixes or relates to

Related to #10188 
Related to https://github.com/prometheus/prometheus/pull/15448

#### Checklist

- [X] Tests updated.
- N/A Documentation added.
- N/A `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- N/A [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by jesusvazquez — APPROVED** [review]

File: pkg/distributor/otel_test.go [hunk 1]
```diff
@@ -50,41 +49,16 @@ func TestOTelMetricsToTimeSeries(t *testing.T) {
 	}, []string{tenantID, "group"})
 	resourceAttrs := map[string]string{
 		"service.name":        "service name",
+		"service.namespace":   "service namespace",
 		"service.instance.id": "service ID",
 		"existent-attr":       "resource value",
 		// This one is for testing conflict with metric attribute.
-		"metric-attr": "resource value",
+		"metric-attr": "resource value conflict",
```
Comment (aknuds1) [inline pkg/distributor/otel_test.go:56]: Is there any practical reason for changing the resource attribute values? I mean, the conflicts are on the attribute names, so I don't think the values need to signal any conflict? The "resource value" part already takes care of signaling that the value comes from a resource attribute.

I also think it ends up giving the wrong impression in test expectations, as it looks as if there's a conflict in `target_info` when there is none.

**Review by aknuds1 — COMMENTED** [review]
Generally looks good, but I don't agree with the changes to the resource attribute values, since I think they give the wrong impression.

File: pkg/distributor/otel_test.go (hunk 1 above)
Comment (krajorama) [reply pkg/distributor/otel_test.go]: Ack, reverted that bit, it wasn't obvious to me what was going on, my change was making it a bit more explicit, but I see your point.

Comment (aknuds1) [reply pkg/distributor/otel_test.go]: Thanks, that's the rationale I suspected :)

**Review by aknuds1 — APPROVED** [review]
LGTM, thanks

**Pull request merged**